### DISCUSSION TITLE
How to add nginx config file that does not get blown away when the rke2 service starts

### DISCUSSION THREAD

**Discussion by sarg3nt** [rancher/rke2#5297] [Q&A]
Hi all.  I'm trying to following the instructions [here](https://docs.rke2.io/networking) by adding a `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml` file during OS creation.  My problem is that during the first boot, when the `rke2` service starts it seems to be blowing away the `/var/lib/rancher/rke2/server/manifests` directory to create its own files.  Thus, my file is deleted and never used.
Is this supposed to happen?
Is there a workaround?

**Comment by brandond** [comment]
You're sure that you're editing `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml` and not `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx.yaml`?  Any files used by default components, including disabled components, will be overwritten or removed on startup.

`/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml` should be a safe filename to use though. Maybe experiment with another name - `/var/lib/rancher/rke2/server/manifests/custom-rke2-ingress-nginx.yaml` for example?
